File: FlexServ Backend/DotNet/AdminService/Models/Address.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AdminService.Models
{
    [Table("addresses")]
    public class Address
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("id")]
        public long Id { get; set; }

        [Column("user_id")]
        public long UserId { get; set; }

        [ForeignKey("UserId")]
        public User? User { get; set; }

        [MaxLength(50)]
        [Column("house_no")]
        public string? HouseNo { get; set; }

        [MaxLength(150)]
        [Column("street")]
        public string? Street { get; set; }

        [MaxLength(100)]
        [Column("area")]
        public string? Area { get; set; }

        [MaxLength(100)]
        [Column("city")]
        public string? City { get; set; }

        [MaxLength(100)]
        [Column("state")]
        public string? State { get; set; }

        [MaxLength(10)]
        [Column("pincode")]
        public string? Pincode { get; set; }
    }
}

File: FlexServ Backend/SpringBoot/TransactionService/src/main/java/com/flexserv/entity/Address.java
package com.flexserv.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "addresses")
public class Address {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(name = "house_no", length = 50)
    private String houseNo;

    @Column(name = "street", length = 150)
    private String street;

    @Column(name = "area", length = 100)
    private String area;

    @Column(name = "city", length = 100)
    private String city;

    @Column(name = "state", length = 100)
    private String state;

    @Column(name = "pincode", length = 10)
    private String pincode;

    public String getFullAddress() {
        StringBuilder sb = new StringBuilder();
        if (houseNo != null && !houseNo.isEmpty()) sb.append(houseNo).append(", ");
        if (street != null && !street.isEmpty()) sb.append(street).append(", ");
        if (area != null && !area.isEmpty()) sb.append(area).append(", ");
        if (city != null && !city.isEmpty()) sb.append(city).append(", ");
        if (state != null && !state.isEmpty()) sb.append(state).append(" - ");
        if (pincode != null && !pincode.isEmpty()) sb.append(pincode);
        return sb.toString().trim();
    }
}
